File: Dante1/Module1.06/Zadanie6.19/main.c
#include <stdio.h>
#include <stdlib.h>

int main()
{
    int matrixA[5][5];
    int matrixB[5][5];

    printf("Enter:\n");

    for(int i = 0; i < 5; i++)
    {
        for(int j = 0; j < 5; j++)
        {
            int check = scanf("%i", &matrixA[i][j]);
            if(check != 1)
            {
                printf("Incorrect input");
                return 1;
            }
        }
    }

    printf("Enter:\n");

    for(int i = 0; i < 5; i++)
    {
        for(int j = 0; j < 5; j++)
        {
            int check = scanf("%i", &matrixB[i][j]);
            if(check != 1)
            {
                printf("Incorrect input");
                return 1;
            }
        }
    }

    int helper[25] = { 0 }; // for storing some numbers


    for(int i = 0; i < 5; i++)
    {
        int counter = 0;

        for(int j = 0; j < 5; j++)
        {
            for(int i1 = 0; i1 < 5; i1++)
            {
                if(matrixA[j][i] == matrixB[i1][i])
                {
                    int temp = 1;
                    for (int j1 = 0; j1 < counter; j1++)
                    {
                        if (helper[j1] == matrixA[j][i])
                        {
                            temp = 0;
                            break;
                        }
                    }

                    if (temp == 1)
                    {
                        helper[counter] = matrixA[j][i];
                        counter++;
                    }
                }
            }
        }

        printf("%i ", counter);

        for(int i1 = 0; i1 < counter; i1++)
        {
            printf("%i ", helper[i1]);
        }
        printf("\n");
    }

    return 0;
}

File: Dante1/Module1.06/Zadanie6.19/main.rs
fn main() {
    let mut input: String = String::new();

    let mut matrix_a: [[i32; 5]; 5] = [[0; 5]; 5];
    println!("Enter: ");
    for i in 0..5 {
        for j in 0..5 {
            input.clear();
            std::io::stdin().read_line(&mut input).expect("Err");
            matrix_a[i][j] = input.trim().parse().expect("err");
        }
    }

    println!("Enter: ");
    let mut matrix_b: [[i32; 5]; 5] = [[0; 5]; 5];
    for i in 0..5 {
        for j in 0..5 {
            input.clear();
            std::io::stdin().read_line(&mut input).expect("Err");
            matrix_b[i][j] = input.trim().parse().expect("err");
        }
    }

    let mut helper: [i32; 25] = [0; 25];
    for i in 0..5 {
        let mut counter: i32 = 0;

        for j in 0..5 {
            for m in 0.. 5 {
                if matrix_a[j][i] == matrix_b[m][i] {
                    let mut temp: i32 = 1;
                    for n in 0..5 {
                        if helper[n] == matrix_a[j][i] {
                            temp = 0;
                            break;
                        }
                    }

                    if temp == 1 {
                        helper[counter as usize] = matrix_a[j][i];
                        counter += 1;
                    }
                }
            }
        }

        print!("{} ", counter);

        for f in 0..counter {
            print!("{} ", helper[f as usize]);
        }
        println!();
    }
}
